cfgov/v1/tests/views/test_reports_view.py

```diff
@@ -0,0 +1,78 @@
+from datetime import date
+
+from django.test import TestCase
+
+from wagtail.core.models import Site
+from wagtail.documents.models import Document
+
+from model_bakery import baker
+from taggit.models import Tag
+
+from v1.models import BlogPage, CFGOVPageCategory
+from v1.util.ref import categories
+from v1.views.reports import (
+    DocumentsReportView, PageMetadataReportView, construct_absolute_url,
+    process_categories, process_tags
+)
+
+
+class ServeViewTestCase(TestCase):
+    def setUp(self):
+        self.page_metadata_report_view = PageMetadataReportView()
+        self.documents_report_view = DocumentsReportView()
+        self.tag1 = baker.make(Tag, name="tag1")
+        self.tag2 = baker.make(Tag, name="tag2")
+        self.site = Site.objects.get(is_default_site=True)
+        self.root_page = self.site.root_page
+        self.blog_page = BlogPage(title="Blogojevich", live=True)
+        self.root_page.add_child(instance=self.blog_page)
+        self.category_tuple = categories[0][1][0]
+        self.category = CFGOVPageCategory(name=self.category_tuple[1])
+        self.blog_page.categories.add(self.category)
+        self.blog_page.tags.add(self.tag1, self.tag2)
+        self.document = Document(title="Test document 1")
+        self.document.save()
+        self.document.tags.add(self.tag1, self.tag2)
+
+    def test_construct_absolute_url(self):
+        self.assertEqual(
+            construct_absolute_url(self.blog_page.url),
+            "https://www.consumerfinance.gov" + self.blog_page.url)
+
+    def test_process_categories(self):
+        category_string = process_categories(self.blog_page.categories.all())
+        self.assertIn(
+            self.category.get_name_display(), category_string)
+        self.assertEqual(
+            self.category.get_name_display(), self.category_tuple[1])
+
+    def test_process_tags(self):
+        tag_name_queryset = self.document.tags.all().values_list(
+            "name", flat=True)
```

Comment:
This is related to the [question I had about what `tags.names` returns in `ReportView`](https://github.com/cfpb/consumerfinance.gov/pull/6900/files#r813255512). `tags.names` gives a queryset of the name of every tag applied to an object in  in `ReportView`, so I'm passing in the same kind of queryset in the test. Is that the best way to do this? Should I add a comment explaining what kind of queryset is expected?

Reply:
I don't think this needs a comment — this follows Django convention as far as I'm aware of it. The only thing is that `all()` should be unnecessary. 
```suggestion
        tag_name_queryset = self.document.tags.values_list(
            "name", flat=True)
```

Reply:
Removed the `.all()` in ef34cdb